content/learn/beginner/control-flow-1.md

```diff
@@ -0,0 +1,213 @@
+
+```go
+{{ if eq 5 3 }}
+  Five is equal to three!
+{{ else }}
+  Five is not equal to three!
+{{ end }}
+```
+
+This can be further expanded to check multiple conditions using `else if`, which are checked sequentially until one of
+them is true:
+
+```go
+{{ if eq 5 3 }}
+  Five is equal to three!
+{{ else if eq 5 5 }}
+  Five is equal to five!
+{{ else }}
+  Five is not equal to three or five!
+{{ end }}
+```
+
+#### Guard Clauses
+
+As your code grows, you may find yourself nesting `if` statements inside each other. This can lead to code that is hard
+to read and understand. One way to avoid this is to use *guard clauses*. A guard clause is an `if` statement that checks
+for a condition and returns early if the condition is false.
+
+Rewriting the second example to use these guard clauses yields the following code:
+
+```go
+{{ if ne 5 3 }}
+  Five is not equal to three!
+  {{ return }}
+{{ end }}
+
+Five is equal to three!
+```
+
+Although this example may be a bit contrived, guard clauses can help you avoid deeply nested code and make your code
+easier to read, especially when you come back to it at a later date.
+
+
+## Blocks And Scope
+
+In [Data Types 1](/learn/beginner/datatypes-1), we introduced the concept of variables. Variables are used to store
+values and give them a name. In programming, variables have a *scope*, which defines where in the code the variable can
+be accessed. Think of each scope as a "container" for things inside it.
+
+Often, you will want to have a variable available across multiple scopes. In Custom Commands, the variable is accessible
+in the scope in which it was defined, as well as all nested scopes within. Let us assume that we want to assign a
+coolness value, which should be true if the user's name is "alice". We can achieve this by defining a variable in
+the outer scope and re-assigning, using the `=` operator, its value in the inner scope:
+
+```go
+{{ $isCool := false }}
+
+{{ if eq .User.Username "alice" }}
+  {{ $isCool = true }}
+{{ end }}
+
+Are you cool? {{ $isCool }}
+```
+
+It is considered good practice to define variables in the smallest scope possible. This makes your code easier to read
+and understand, as you do not have to search through the entire codebase to find where a variable was defined.
+
+{{< callout context="caution" title="Definition and Reassignment" icon="outline/alert-triangle" >}}
+
+In Custom Commands, you use `:=` to define a variable, and `=` to reassign a variable. The bot will not throw an error
+if you try to re-define a variable using `:=`, but it will not affect the outer scoped variable.
+
+{{< /callout >}}
+
+## Boolean Logic
+
+To form more complex conditional checks, we need to understand *boolean logic*. [Boolean logic] is a branch of algebra
+that deals with true/false values. In programming, we use boolean logic to make decisions based on the truth value of
+expressions. There are three primary boolean operators: `AND`, `OR`, and `NOT`.
```

Comment:
Mentioning boolean algebra is misplaced in an introductory document. All the operators have natural explanations: e.g., OR reports whether at least one expression is true: I will be late to school if `(i wake up late) or (the bus is late)`.

Reply:
It was there at the start of the old learn resources 😎. I do agree
though, it's a bit of a jump to go from simple comparisons to full on
boolean algebra. Do we want to place that in a reference document?